### PR TITLE
Print retires & display additional info

### LINKED ISSUE #717: Print retries in Flank so that the user knows what's happening
As reported in #flank on FTL Slack.

> [task 2020-04-13T20:47:55.150Z]   5m 43s matrix-3ru2up8r1oh9z Pixel2-28 FINISHED
> [task 2020-04-13T20:50:31.384Z]   8m 20s matrix-3ru2up8r1oh9z FINISHED

We should print to stdout on each retry.

### PR DESCRIPTION
**Pull request by jan-gogo**
Fixes #717 #681 #738 

## Checklist

- [x] Unit tested
- [x] release_notes.md updated

### CODE REVIEW

**Comment by jan-gogo** [comment]
```
AndroidArgs
    gcloud:
      results-bucket: test-lab-v9cn46bb990nx-kz69ymd4nm9aq
      results-dir: null
      record-video: false
      timeout: 15m
      async: false
      client-details: 
      network-profile: null
      results-history-name: null
      # Android gcloud
      app: /Users/janek/projects/flank-project/flank/test_runner/src/test/kotlin/ftl/fixtures/tmp/apk/app-debug.apk
      test: /Users/janek/projects/flank-project/flank/test_runner/src/test/kotlin/ftl/fixtures/tmp/apk/app-multiple-flaky-debug-androidTest.apk
      additional-apks: 
      auto-google-login: false
      use-orchestrator: true
      directories-to-pull:
        - /sdcard/
      other-files:
      performance-metrics: false
      num-uniform-shards: null
      test-runner-class: null
      test-targets:
      robo-directives:
      robo-script: null
      device:
        - model: NexusLowRes
          version: 28
          locale: en
          orientation: portrait
      num-flaky-test-attempts: 3

    flank:
      max-test-shards: 2
      shard-time: -1
      num-test-runs: 1
      smart-flank-gcs-path: 
      smart-flank-disable-upload: false
      files-to-download:
        - .*/sdcard/[^/]+\.ec$
      test-targets-always-run:
      disable-sharding: false
      project: flank-open-source
      local-result-dir: results
      # Android Flank Yml
      keep-file-path: false
      additional-app-test-apks:
      run-timeout: -1
      legacy-junit-result: false
      ignore-failed-tests: false

RunTests

  Smart Flank cache hit: 0% (0 / 5)
  Shard times: 240s, 360s

  Uploading app-multiple-flaky-debug-androidTest.apk   Uploading app-debug.apk ..

  5 tests / 2 shards

  1 matrix ids created in 0m 7s
  https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/2020-05-12_12-33-23.713000_vrEO/

Matrices webLink
  matrix-bxs7irgqf2mda https://console.firebase.google.com/project/flank-open-source/testlab/histories/bh.da0c237aaa33732/matrices/7945200213789677367

  0m  1s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd PENDING
  0m  1s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h PENDING
  0m 32s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Starting attempt 1.
  0m 32s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd RUNNING
  0m 32s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Starting attempt 1.
  0m 32s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h RUNNING
  1m 22s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Started logcat recording.
  1m 22s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Preparing device.
  1m 28s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Started logcat recording.
  1m 28s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Preparing device.
  1m 46s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Installing apps.
  1m 46s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Installing apps.
  1m 59s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Retrieving Pre-Test Package Stats information from the device.
  1m 59s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Retrieving Performance Environment information from the device.
  1m 59s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Started crash detection.
  1m 59s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Started crash monitoring.
  1m 59s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Starting instrumentation test.
  2m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Retrieving Pre-Test Package Stats information from the device.
  2m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Retrieving Performance Environment information from the device.
  2m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Started crash detection.
  2m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Started crash monitoring.
  2m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Starting instrumentation test.
  2m 11s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Completed instrumentation test.
  2m 11s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Completed instrumentation test.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa PENDING
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Stopped crash monitoring.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Retrieving Post-test Package Stats information from the device.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Stopped logcat recording.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Done. Test time = 14 (secs)
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Starting results processing. Attempt: 1
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd Completed results processing. Time taken = 2 (secs)
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-22t43h1j7e2wd FINISHED
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr PENDING
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Stopped crash monitoring.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Retrieving Post-test Package Stats information from the device.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Stopped logcat recording.
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Done. Test time = 10 (secs)
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Starting results processing. Attempt: 1
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h Completed results processing. Time taken = 2 (secs)
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-0 execution-2ppaxo6r1ve1h FINISHED
  2m 35s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u PENDING
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Starting attempt 1.
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa RUNNING
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Starting attempt 1.
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr RUNNING
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Starting attempt 1.
  3m  0s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u RUNNING
  3m 49s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Started logcat recording.
  3m 49s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Preparing device.
  3m 49s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Started logcat recording.
  3m 49s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Preparing device.
  4m  2s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Started logcat recording.
  4m  2s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Preparing device.
  4m 15s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Installing apps.
  4m 15s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Installing apps.
  4m 15s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Retrieving Pre-Test Package Stats information from the device.
  4m 15s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Retrieving Performance Environment information from the device.
  4m 27s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Retrieving Pre-Test Package Stats information from the device.
  4m 27s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Retrieving Performance Environment information from the device.
  4m 27s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Started crash detection.
  4m 27s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Started crash monitoring.
  4m 27s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Starting instrumentation test.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Started crash detection.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Started crash monitoring.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Starting instrumentation test.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Installing apps.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Retrieving Pre-Test Package Stats information from the device.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Retrieving Performance Environment information from the device.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Started crash detection.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Started crash monitoring.
  4m 34s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Starting instrumentation test.
  4m 40s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Completed instrumentation test.
  4m 40s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Completed instrumentation test.
  4m 53s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Completed instrumentation test.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Stopped crash monitoring.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Retrieving Post-test Package Stats information from the device.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Stopped logcat recording.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Done. Test time = 13 (secs)
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Starting results processing. Attempt: 1
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa Completed results processing. Time taken = 2 (secs)
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-15nt5nhl4twoa FINISHED
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Stopped crash monitoring.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Retrieving Post-test Package Stats information from the device.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Stopped logcat recording.
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Done. Test time = 15 (secs)
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Starting results processing. Attempt: 1
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr Completed results processing. Time taken = 2 (secs)
  5m  5s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-2ohsc9p6f57pr FINISHED
  5m 11s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Stopped crash monitoring.
  5m 11s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Retrieving Post-test Package Stats information from the device.
  5m 18s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Stopped logcat recording.
  5m 18s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Done. Test time = 13 (secs)
  5m 18s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Starting results processing. Attempt: 1
  5m 24s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u Completed results processing. Time taken = 2 (secs)
  5m 24s matrix-bxs7irgqf2mda NexusLowRes-28 shard-1 execution-3gvs6r96igw5u FINISHED
  5m 24s matrix-bxs7irgqf2mda FINISHED
FetchArtifacts
  .....
  Updating matrix file

CostReport
  Virtual devices
    $0.08 for 5m

MatrixResultsReport
  0 / 1 (0.00%)
  1 matrices failed

Error: Matrix failed: matrix-bxs7irgqf2mda FINISHED failure  https://console.firebase.google.com/project/flank-open-source/testlab/histories/bh.da0c237aaa33732/matrices/7945200213789677367

Process finished with exit code 1

```

**Comment by jan-gogo** [comment]
I have added 2 additional styles for displaying single & multi line shortened output. On consoles which not honors ansi codes multi line output will not be display correctly so single line output can be used. 

Multi line output: 
```
  3m  1s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-0 execution-1745swg41t0b2 FINISHED
  2m 55s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-1 execution-2tgcqxrrupgu5 FINISHED
  4m 58s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-1 execution-1689yp4w2xewa Starting instrumentation test.
  4m 23s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-0 execution-17j3zh3bwvc5m Preparing device.
  4m 52s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-0 execution-28h3aflk56m55 Installing apps.
  4m 58s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-0 execution-2fht77lyehkk8 Completed instrumentation test.
  4m 52s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-1 execution-3ttu7s12ylovo Starting instrumentation test.
  4m 58s matrix-3hy98fkuzmdpy NexusLowRes-28 shard-1 execution-ugdjrkzws8mka Starting instrumentation test.
```

Single line output:
```
3m 35s Test executions status: FINISHED:2, PENDING:2 RUNNING:4
```
In both cases old output is overwritten when status has changed
Single line output is a port from gcloud `MonitorTestMatrixProgress` function

I will add new flank option `output_style` to allow force specific style.
And I have to add some unit tests also.

File: test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+@VisibleForTesting
+internal object VerbosePrinter : (List<ExecutionStatus.Change>) -> Unit {
+    override fun invoke(changes: List<ExecutionStatus.Change>) {
+        changes.map(listExecutionStatusView).flatten().forEach(::println)
+    }
+}
+
+private val listExecutionStatusView: (ExecutionStatus.Change) -> List<ExecutionStatus.View>
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt:68]: How about extension property here instead? Something like (just proposal, not change request :) ):
```
private val ExecutionStatus.Change.views: List<ExecutionStatus.View>
    get() = current.progress
        .minus(previous.progress)
        .map { ExecutionStatus.View(time, name, it) }
        .let { list ->
            if (current.error != previous.error && current.error != null)
                list + ExecutionStatus.View(time, name, "Error: ${current.error}")
            else list
        }
        .let { list ->
            if (current.state != previous.state)
                list + ExecutionStatus.View(time, name, current.state)
            else list
        }
```

Let me know what you think?

File: test_runner/src/main/kotlin/ftl/run/status/TestMatrixStatusPrinter.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private val args: IArgs,
+    testMatricesIds: Iterable<String>,
+    private val stopWatch: StopWatch = StopWatch(),
+    private val printExecutionStatusList: (String, List<TestExecution>?) -> Unit = ExecutionStatusListPrinter(args)
+) : (TestMatrix) -> Unit {
+    init {
+        stopWatch.start()
+    }
+
+    private val cache = testMatricesIds.map { it to MatrixState.PENDING }.toMap().toMutableMap()
```
[the remark this answers is not in the document]
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/status/TestMatrixStatusPrinter.kt:21]: Nice, I missed this, thx.

File: test_runner/src/main/kotlin/ftl/run/common/PollMatrices.kt [hunk 3]
```diff
@@ -1,64 +1,54 @@
...
-        if (runningDevices.allComplete() && refreshedMatrix.completed()) {
-            break
-        }
-
-        // GetTestMatrix is not designed to handle many requests per second.
-        // Sleep to avoid overloading the system.
-        delay(5_000)
-        refreshedMatrix = GcTestMatrix.refresh(matrixId, args.project)
+        val matrix = GcTestMatrix.refresh(testMatrixId, projectId)
+        emit(matrix)
```
[the remark this answers is not in the document]
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/common/PollMatrices.kt:47]: Maybe you have right but is should be done by `GcTestMatrix.refresh`, not here

File: test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt (hunk 1 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt:68]: It doesn't make much difference, but I can change it.

File: test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusListPrinter.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            },
+            time = time
+        )
+    } ?: emptyList()
+}
+
+private fun TestExecution.formatName(args: IArgs): String {
+    val matrixExecutionId = id.split("_")
+    val matrixId = matrixExecutionId.first()
+    val executionId = matrixExecutionId.takeIf { args.flakyTestAttempts > 0 }?.getOrNull(1)?.let { " $it" } ?: ""
```
[the remark this answers is not in the document]
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusListPrinter.kt:35]: I agree, in the future. ;)

File: test_runner/src/main/kotlin/ftl/run/NewTestRun.kt [hunk 5]
```diff
@@ -20,7 +21,7 @@ suspend fun newTestRun(args: IArgs) {
         val (matrixMap, testShardChunks) = cancelTestsOnTimeout(args.project) { runTests(args) }
 
         if (!args.async) {
-            cancelTestsOnTimeout(args.project, matrixMap.map) { pollMatrices(matrixMap, args) }
+            cancelTestsOnTimeout(args.project, matrixMap.map) { pollMatrices(matrixMap.map.keys, args).update(matrixMap) }
```
[the remark this answers is not in the document]
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/NewTestRun.kt:24]: The idea was that pollMatrices shoud not be aware of SavedMatrix. SavedMatrix looks like old school java class. I dont like black boxes. I have plans to get rid of it in the future.

File: test_runner/src/main/kotlin/ftl/run/status/TestMatrixStatusPrinter.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    private val allMatricesCompleted get() = cache.values.all(MatrixState::completed)
+
+    override fun invoke(matrix: TestMatrix) {
+        val time = stopWatch.check(alignSeconds = true)
+        printExecutionStatusList(time, matrix.testExecutions)
+        cache[matrix.testMatrixId] = matrix.state
+        if (allMatricesCompleted) printTestMatrixStatusList(time)
```
[the remark this answers is not in the document]
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/status/TestMatrixStatusPrinter.kt:28]: I will think about that but pls, not in this PR.

File: test_runner/src/main/kotlin/ftl/run/NewTestRun.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (pawelpasterz) [reply test_runner/src/main/kotlin/ftl/run/NewTestRun.kt:24]: Makes sens, thanks.

File: test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt (hunk 1 above)
Comment (pawelpasterz) [reply test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt]: IMO it's more clear when I see sth like
`change.views.takeLast`
but I don't want to force anything.

File: release_notes.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ## next (unreleased)
 
+- [#779](https://github.com/Flank/flank/pull/779) Print retires & display additional info. ([jan-gogo](https://github.com/jan-gogo))
```
Comment (runningcode) [inline release_notes.md:3]: I wish I could retire early 🤣 
```suggestion
- [#779](https://github.com/Flank/flank/pull/779) Print retries & display additional info. ([jan-gogo](https://github.com/jan-gogo))
```

**Pull request merged**